hot-reload.md

```diff
@@ -0,0 +1,279 @@
+---
+layout: page
+title: Hot Reloading Flutter Apps
+permalink: /hot-reload/
+---
+
+* TOC
+{:toc}
+
+## Using hot reload
+
+Flutter's hot reload feature helps you quickly and easily experiment, build
+UIs, add features and fix bugs. Hot reload works by injecting updated source
+code files into the running Dart Virtual Machine (VM). After the VM updates
+classes with the new versions of fields and functions, the Flutter framework
+automatically rebuilds the widget tree, allowing you to quickly view the effects
+of your changes.
+
+To hot reload a Flutter app:
+
+1.  Run the app from a supported IntelliJ IDE or a terminal window. Either a
+physical or virtual device can be the target.
+1.  Modify one of the Dart files in your project. Most types of code changes can
+be hot reloaded; for a list of changes that require a full restart, see
+[Understanding limitations](#understanding-limitations).
+1.  If you're working in an IntelliJ IDE that supports Flutter's IDE tools,
+click the Hot Reload button on the toolbar:
+
+![alt_text](/images/intellij/hot-reload.gif "image_tooltip")
+
+If you're running the app at the command line using `flutter run`, type `r` in
+the terminal window. 
+
+After a successful hot reload operation, you'll see a message in the console
+similar to:
+
+```
+Performing hot reload...
+Reloaded 1 of 448 libraries in 2,777ms.
+```
+The app's UI is updated to reflect your change, and the current state of the app
+— the value of the counter variable in the above example — is preserved. Your
+app continues to execute from where it was prior to running the hot reload
+command. The code is updated and execution continues.
+
+## Troubleshooting hot reload
+
+When you start working with hot reload and using it frequently, you'll likely
+encounter the types of code changes that might need a full app restart:
+
+
+
+*   Initializing global variables
```

Comment:
It would be good to synchronize this list with the examples given below. For example, `main()` is third on the list here but first in the detail section below.

Reply:
Good point, done!